Next

mhartington opened the pull request:
This PR introduces several new features to angular toolkit. 

1) bumps the minimum version support to Angular v15
2) support for generating standalone pages
3) updates the page template to use the large header/title layout

liamdebeasi commented on packages/schematics/page/route-utils/index.ts:
```diff
@@ -0,0 +1,253 @@
+    if (matches.length === 1) {
+      return join(dir.path, matches[0]);
+    } else if (matches.length > 1) {
+      throw new Error('whoops');
```
Can we add a more descriptive error message here?

liamdebeasi commented on packages/schematics/page/route-utils/index.ts:
```diff
@@ -0,0 +1,253 @@
+    }
+    dir = dir.parent;
+  }
+  throw new Error('Could not find your routes file. Use the skip-import ' + 'option to skip importing.');
```
Why do we do ` ' + '`?

liamdebeasi commented on packages/schematics/page/route-utils/index.ts:
```diff
@@ -0,0 +1,253 @@
+      return join(dir.path, matches[0]);
+    } else if (matches.length > 1) {
+      throw new Error(
+        'More than one module matches. Use skip-import option to skip importing ' +
```
Why do we do ` ' + '`?

mhartington commented:
>Why do we do  ' + '?

🤷  Some of this code predates me maintaining angular-toolkit, but I think it's safe to remove.

liamdebeasi reviewed (dismissed):
Code looks good, but I added Sean to take a look too.

Only final thing is can we add a write-up for the PR as to what it does/why it's needed?

mhartington commented:
Rebased and cleaned up the commits so we have can have 3 separate commits that are clear and understood. This should not be squashed, but merged in the traditional way
